packages/turf-boolean-crosses/index.js

```diff
@@ -0,0 +1,153 @@
+var helpers = require('@turf/helpers');
+var inside = require('@turf/inside');
+var lineIntersect = require('@turf/line-intersect');
+var polyToLinestring = require('@turf/polygon-to-linestring');
+var invariant = require('@turf/invariant');
+var getGeom = invariant.getGeom;
+var getGeomType = invariant.getGeomType;
+
+/**
+ * Boolean-Crosses returns True if the intersection results in a geometry whose dimension is one less than
+ * the maximum dimension of the two source geometries and the intersection set is interior to
+ * both source geometries.
+ *
+ * Boolean-Crosses returns t (TRUE) for only multipoint/polygon, multipoint/linestring, linestring/linestring, linestring/polygon, and linestring/multipolygon comparisons.
+ *
+ * @name booleanCrosses
+ * @param {Geometry|Feature<any>} feature1 GeoJSON Feature or Geometry
+ * @param {Geometry|Feature<any>} feature2 GeoJSON Feature or Geometry
+ * @returns {Boolean} true/false
+ * @example
+ * var cross = turf.booleanCrosses(feature1, feature2);
+ */
+
+module.exports = function (feature1, feature2) {
+    var type1 = getGeomType(feature1);
+    var type2 = getGeomType(feature2);
+    var geom1 = getGeom(feature1);
+    var geom2 = getGeom(feature2);
+
+    switch (type1) {
+    case 'MultiPoint':
+        switch (type2) {
+        case 'LineString':
+            return doMultiPointAndLineStringCross(geom1, geom2);
+        case 'Polygon':
+            return doesMultiPointCrossPoly(geom1, geom2);
+        }
+        throw new Error('feature2 ' + type2 + ' geometry not supported');
+    case 'LineString':
+        switch (type2) {
+        case 'LineString':
+            return doLineStringsCross(geom1, geom2);
+        case 'Polygon':
+            return doLineStringAndPolygonCross(geom1, geom2);
+        }
+        throw new Error('feature2 ' + type2 + ' geometry not supported');
+    default:
+        throw new Error('feature1 ' + type1 + ' geometry not supported');
+    }
+};
+
+function doMultiPointAndLineStringCross(multiPoint, lineString) {
+    var foundIntPoint = false;
+    var foundExtPoint = false;
+    var pointLength = multiPoint.coordinates.length;
+    var i = 0;
+    while (i < pointLength && !foundIntPoint && !foundExtPoint) {
+        for (var i2 = 0; i2 < lineString.coordinates.length - 1; i2++) {
+            var incEndVertices = true;
+            if (i2 === 0 || i2 === lineString.coordinates.length - 2) {
+                incEndVertices = false;
+            }
+            if (isPointOnLineSegment(lineString.coordinates[i2], lineString.coordinates[i2 + 1], multiPoint.coordinates[i], incEndVertices)) {
```

Comment:
actually now I remember why I did this, I worked out that this didn't matter as the actual 2nd vertex get's picked up in the second segment.